### PR TITLE
Pull in conor's scripts

### PR DESCRIPTION
**Pull request by clmorrison**
split files, made package loading more robust, also saves an RDS of the formatted metadata for use in my scripts (gets saved into results).

### CODE REVIEW

**Comment by creisle** [comment]
unless someone else is already on it i'm going to try to add these to the snakemake

File: src/kmeans.R [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+}
+
+
+fit_dat=readRDS("./results/metadata_formatted.rds")
+km_dat=filter(fit_dat,tumor) %>%
+  filter(subtype!="none")
+km_res=kmeans(km_dat[,18:dim(train)[2]], 4, iter.max = 10, nstart = 5)
+
+
+jpeg('./results/PC_survplot.jpg',height=150,width = 200)
```
Comment (creisle) [inline src/kmeans.R:21]: The name of the output here is a bit confusing. This is the survival plot based on the kmeans groupings? 

longer, more descriptive names are better if you have time to rename :)

File: src/metadata_formatting.R [hunk 2]
```diff
@@ -0,0 +1,128 @@
+
+
+
+
+dat=read.table("./data/crc_b_crrctd_low_rmvd.txt",stringsAsFactors = F, header = T, sep = "\t")
+metadat=read.table("./data/COADREAD_metadata.txt",stringsAsFactors = F, header = T, sep = "\t")
+subtypedat=read.table("./results/subtype_svc.classification.txt",stringsAsFactors = F, header = T, sep = "\t")
```
Comment (creisle) [inline src/metadata_formatting.R:7]: I think the filename is wrong here? this should be .tsv

Comment (clmorrison) [reply src/metadata_formatting.R:7]: Oh my bad I had issues downloading the tsv it got saved as a txt.

It **should** be .tsv.

Good catch!

File: src/kmeans.R (hunk 1 above)
Comment (creisle) [reply src/kmeans.R:21]: @clmorrison looks like this file name is output from 2 diff scripts. Could you change it to add something like

`results/survival.PC.kmeans.jpg` or similar? if all your survival results start with `survival` (or other common prefix) it makes them group/view nicer

**Review by creisle — DISMISSED** [review]
see above comments for requested changes (only fixing the file naming conflict is required)

File: src/kmeans.R (hunk 1 above)
Comment (clmorrison) [reply src/kmeans.R:21]: Oh total mistake on my part. I copied and pasted that line from the other script but forgot to change the actual name. Let's name it km_table

**Comment by clmorrison** [comment]
Ok I 

* changed the txt to tsv when loading the subtype data
* changed the image name for the kmeans

**Comment by creisle** [comment]
> Ok I
> 
> * changed the txt to tsv when loading the subtype data
> * changed the image name for the kmeans

Did you also update the name in the snakemake to match?

**Comment by clmorrison** [comment]
> > Ok I
> > 
> > * changed the txt to tsv when loading the subtype data
> > * changed the image name for the kmeans
> 
> Did you also update the name in the snakemake to match?

I was about to do it once you mentioned but I think it said you updated it

**Comment by creisle** [comment]
@clmorrison can you revert your last commit and pull the branch first? it looks like you overwrote all my snakemake fixes (using `here` for all paths)

**Comment by creisle** [comment]
> @clmorrison can you revert your last commit and pull the branch first? it looks like you overwrote all my snakemake fixes (using `here` for all paths)

going to try to push a fix for this. its a bit complicated to fix

**Review by creisle — APPROVED** [review]

**Pull request merged**